### ISSUE TITLE
2020

### ISSUE THREAD

**Issue by Bahubali120** [Bahubali120/KA-49#5]
import tensorflow as tf
from tensorflow import keras
import os,datetime

Mnist = tf.keras.datasets.mnist
(x_train,y_train),(x_test,y_test)=Mnist.load_data()
print("training dataset shape:",x_train.shape)
print("no of training dataset samples:",len(x_test))
print("testing dataset shape:",x_test.shape)
print("no of testing dataset samples:",len(y_test))

import matplotlib.pyplot as plt
plt.imshow(x_train[3])
plt.show()
print("first digits in y train is",y_train[3])

x_train,x_test=x_train/255.0,x_test/255.0
def create_model():
  model=tf.keras.models.Sequential([
      tf.keras.layers.Flatten(input_shape=(28,28)),
      tf.keras.layers.Dense(512,activation='relu'),
      tf.keras.layers.Dropout(0.2),
      tf.keras.layers.Dense(10,activation='softmax')
  ])
  return model

  model=create_model()
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

logdir=os.path.join("logs",datetime.datetime.now().strftime("%Y%m%d-%H%M%S"))
tensorboard_callback=tf.keras.callbacks.TensorBoard(logdir,histogram_freq=1)

checkpoint_path="Model_checkpoint/cp-{epoch:04d}.weights.h5"
checkpoint_dir=os.path.dirname(checkpoint_path)
checkpoint_callback=tf.keras.callbacks.ModelCheckpoint(checkpoint_path,monitor='val_acc',verbose=1,save_weights_only=True,save_freq='epoch')

history=model.fit(x_train,y_train,epochs=5,validation_data=(x_test,y_test),
                  callbacks=[tensorboard_callback,checkpoint_callback])
loss,acc=model.evaluate(x_test,y_test,verbose=1)
print("originals model, accuracy:{:5.2f}%".format(100*acc))

%load_ext tensorboard

tensorboard --logdir=logs --port=6007
